yellowbrick/cluster/elbow.py

```diff
@@ -262,6 +270,9 @@ def fit(self, X, y=None, **kwargs):
                 self.scoring_metric(X, self.estimator.labels_)
             )
 
+        self.kneedle=KneeLocator(self.k_values_,self.k_scores_,curve='convex',direction='decreasing')
+        self.knee_value=self.kneedle.find_knee()[0]
+        self.score=self.k_scores_[self.k_values_.index(self.knee_value)]
```

Comment:
So one reason a user might make `locate_elbow=False` is if the `KneeLocator` is throwing an exception of some kind or is taking an unreasonably long time to run. These lines of code run no matter the `locate_elbow` flag - so might I recommend the following?

```python
        if self.locate_elbow:
            knee_locator = KneeLocator(...)
            self.elbow_value_ = knee_locator.find_knee()[0]
            self.elbow_score_ = self.k_scores_[self.k_values_.index(self.elbow_value_)]
        else:
            self.elbow_value_ = None
            self.elbow_score_ = None
```

This also gives information to developers about the learned state of the visualizer.

Reply:
Oh! sorry, that's definetly a good pointer. I missed that. I'll update that too.

Reply:
No need to apologize, we don't expect or require perfect code in a PR - that's what these reviews are for, so we can discuss how to make the code as excellent as possible.